siliconcompiler/schema/schema_cfg.py

```diff
@@ -3507,96 +3595,185 @@ def schema_asic(cfg):
             sctype='str',
             scope='job',
-            shorthelp="Constraint operating mode",
-            switch="-constraint_mode 'scenario <str>'",
-            example=["cli: -constraint_mode 'worst test'",
-                     "api: chip.set('constraint',  'worst','mode', 'test')"],
+            shorthelp="Constraint: operating mode",
+            switch="-constraint_timing_mode 'scenario <str>'",
+            example=["cli: -constraint_timing_mode 'worst test'",
+                     "api: chip.set('constraint', 'timing', 'worst','mode', 'test')"],
             schelp="""Operating mode for the scenario. Operating mode strings
             can be values such as test, functional, standby.""")
 
-    scparam(cfg,['constraint', scenario, 'file'],
+    scparam(cfg,['constraint', 'timing', scenario, 'file'],
             sctype='[file]',
             scope='job',
             copy='true',
-            shorthelp="Constraint files",
-            switch="-constraint_file 'scenario <file>'",
-            example=["cli: -constraint_file 'worst hello.sdc'",
-                     "api: chip.set('constraint','worst','file', 'hello.sdc')"],
+            shorthelp="Constraint: SDC files",
+            switch="-constraint_timing_file 'scenario <file>'",
+            example=[
+                "cli: -constraint_timing_file 'worst hello.sdc'",
+                "api: chip.set('constraint', 'timing', 'worst','file', 'hello.sdc')"],
             schelp="""List of timing constraint files to use for the scenario. The
             values are combined with any constraints specified by the design
             'constraint' parameter. If no constraints are found, a default
             constraint file is used based on the clock definitions.""")
 
-    scparam(cfg,['constraint', scenario, 'check'],
+    scparam(cfg,['constraint', 'timing', scenario, 'check'],
             sctype='[str]',
             scope='job',
-            shorthelp="Constraint checks to perform",
-            switch="-constraint_check 'scenario <str>'",
-            example=["cli: -constraint_check 'worst check setup'",
-                    "api: chip.add('constraint','worst','check','setup')"],
+            shorthelp="Constraint: timing checks",
+            switch="-constraint_timing_check 'scenario <str>'",
+            example=[
+                "cli: -constraint_timing_check 'worst check setup'",
+                "api: chip.add('constraint', 'timing', 'worst','check','setup')"],
             schelp="""
             List of checks for to perform for the scenario. The checks must
             align with the capabilities of the EDA tools and flow being used.
             Checks generally include objectives like meeting setup and hold goals
             and minimize power. Standard check names include setup, hold, power,
             noise, reliability.""")
 
+    # COMPONENTS
+    scparam(cfg, ['constraint', 'component', instance, 'outline'],
```

Comment:
What is the intent of this parameter? It's not clear to me what an outline constraint means?

Reply:
It looks like a way to represent a macro's size in the schema? We may be able to get those values from the `('asic', 'diearea')` parameter in the macro's schema, or the LEF file if the designs were auto-sized and/or built outside of SC.

Although, the `diearea` parameter doesn't support rectilinear shapes, and this may be intended to represent components in other contexts besides asic physical design flows.